src/openqaoa-braket/openqaoa_braket/backends/devices.py

```diff
@@ -144,7 +145,23 @@ def _check_provider_connection(self) -> bool:
             )
             return False
 
+    @property
     def connectivity(self) -> List[List[int]]:
-        aws_connectivity = self.backend_device.topology_graph.edges
-        connectivity_list = [list(edge) for edge in aws_connectivity]
-        return connectivity_list
+        aws_device_properties = self.backend_device.properties
+        if hasattr(aws_device_properties, "topology_graph"):
+            aws_device_topology = aws_device_properties.topology_gprah
+            if aws_device_topology is not None:
+                aws_connectivity = aws_device_topology.edges
+                connectivity_list = [list(edge) for edge in aws_connectivity]
```

Comment:
I guess we could remove a few lines and simply have

`self._connectivity = [list(edge) for edge in aws_device_topology.edges] `

Reply:
fair, this happens when you are too focused on catching errors haha

Reply:
done